test/function/samples/resolveid-recursive-call/_config.js

```diff
@@ -0,0 +1,29 @@
+module.exports = defineTest({
+	description:
+		'skipSelf: true option in resolveId hook option should skip the plugin if it has been called before with the same id and importer, see #5768 for more details',
+	options: {
+		plugins: [
+			{
+				name: 'r1',
+				resolveId(id) {
+					console.log('r1', id);
+					const importer = 'foo';
+					return this.resolve(id, importer) ?? 'success';
+				},
+				load(id) {
+					if (id === 'success') {
+						return { code: 'export default 1' };
+					}
+				}
+			},
+			{
+				name: 'r2',
+				resolveId(id) {
+					console.log('r2', id);
```

Comment:
Please do not `console.log` in tests as this spams the test output. While this is nice for debugging, if you want to check which hooks are called, why not instead push values into an array and at the end of the test, assert which calls happened in which order? This would immediately show what the intended outcome is and possibly even improve the test.